syntax = "proto3";

message Input {
    string model = 1;
    string img_path = 2;
    string confidence = 3;
}

message Boxes {
    float x = 1;
    float y = 2;
    int32 w = 3;
    int32 h = 4;
}

message Output {
    float delta_time = 1;
    repeated Boxes boxes = 2;
    repeated int32 class_ids = 3;
    repeated float confidences = 4;
    bytes img_base64 = 5;
}

service Detect {
    rpc detect(Input) returns (Output) {}
}
